src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -0,0 +1,305 @@
+                    {
+                        Logger.Message("Metrics filter argument received: " + metricsSpecs);
+                        ParseSpecs(metricsSpecs);
+                    }
+                    else
+                    {
+                        Logger.Message("No Metrics filter argument received");
+                    }
+
+                    _aggregationManager.Start();
+                }
+            }
+
+            private static char[] s_instrumentSeperators = new char[] { '\r', '\n', ',', ';' };
+
+            private void ParseSpecs(string? metricsSpecs)
+            {
+                if (metricsSpecs == null)
+                {
+                    return;
+                }
+                string[] specStrings = metricsSpecs.Split(s_instrumentSeperators, StringSplitOptions.RemoveEmptyEntries);
+                foreach (string specString in specStrings)
+                {
+                    if (!MetricSpec.TryParse(specString, out MetricSpec spec))
+                    {
+                        Logger.Message("Failed to parse metric spec: " + specString);
+                    }
+                    else
+                    {
+                        Logger.Message("Parsed metric: " + spec.ToString());
+                        if (spec.InstrumentName != null)
+                        {
+                            _aggregationManager!.Include(spec.MeterName, spec.InstrumentName);
+                        }
+                        else
+                        {
+                            _aggregationManager!.Include(spec.MeterName);
+                        }
+                    }
+                }
+            }
+
+            private void TransmitMetricValue(Instrument instrument, LabeledAggregationStatistics stats)
+            {
+                if (stats.AggregationStatistics is RateStatistics rateStats)
+                {
+                    Logger.CounterRateMetricPublished(instrument.Meter.Name, instrument.Meter.Version, instrument.Name, instrument.Unit, FormatTags(stats.Labels), rateStats.Delta);
+                }
+                else if (stats.AggregationStatistics is LastValueStatistics lastValueStats)
+                {
+                    Logger.GaugeMetricPublished(instrument.Meter.Name, instrument.Meter.Version, instrument.Name, instrument.Unit, FormatTags(stats.Labels), lastValueStats.LastValue);
+                }
+                else if (stats.AggregationStatistics is HistogramStatistics histogramStats)
+                {
+                    Logger.HistogramMetricPublished(instrument.Meter.Name, instrument.Meter.Version, instrument.Name, instrument.Unit, FormatTags(stats.Labels), FormatQuantiles(histogramStats.Quantiles));
+                }
+            }
+
+            private string FormatTags(KeyValuePair<string, string>[] labels)
+            {
+                StringBuilder sb = new StringBuilder();
```

Comment:
Should this be `ValueStringBuilder` to avoid the allocation(s) or doesn't it matter here?

Reply:
I expect it not to matter. This code path runs each collection interval which is typically 1 sec - 10 minutes in length. Given the infrequency I biased towards simple code.